cloudify_aws/autoscaling/resources/launch_configuration.py

```diff
@@ -97,9 +101,9 @@ def create(ctx, iface, resource_config, params, **_):
     # Check if the "IamInstanceProfile" is passed or not and then update it
     iam_instance_profile = params.get(IAM_INSTANCE_PROFILE)
     if iam_instance_profile:
-        if isinstance(iam_instance_profile, basestring):
+        if isinstance(iam_instance_profile, text_type):
             iam_instance_profile = iam_instance_profile.strip()
-            params[IAM_INSTANCE_PROFILE] = str(iam_instance_profile)
+            params[IAM_INSTANCE_PROFILE] = text_type(iam_instance_profile)
```

Comment:
Well, since we need to make everything as unicode, then I used the text_type so on python 3 the `str()` will be unicode and on python 2 it will use `unicode` so want to be consistent on both versions

Reply:
I'll make a jira to revisit it.